### DISCUSSION TITLE
[Question]attachment_can_read behavior

### DISCUSSION THREAD

**Discussion by tohort** [django-wiki/django-wiki#1268] [Q&A]
In plugins/attachments/markdown_extensions.py Line77
`attachment_can_read = can_read(self.markdown.article, article_owner)`

this check the current article can be read by the attachement owner.
However, I think it should be the following. (this check the atacchement can be read by access user)
`attachment_can_read = can_read(attachment.article, self.request.user)`

There is any reason for this?

**Comment by benjaoming** [comment]
I think it's correct given the full context. This is happening in the rendering of Markdown, so there is only one relevant question: When we render an attachment (link!) in an article, we want to know that the owner of the article can actually read the attachment. Otherwise, you could use this to "escalate" privileges to read attachments through articles that you own.

https://github.com/django-wiki/django-wiki/blob/0e25c3eea3e2608aa3be7ebd5dee8a9faf8dc131/src/wiki/plugins/attachments/markdown_extensions.py#L64-L75
